### ISSUE TITLE
Message Content Type

### ISSUE THREAD

**Issue by cvarjao** [hyperledger/aries-rfcs#786]
Bases on Aries [RFC 0095: Basic Message Protocol 1.0](https://github.com/hyperledger/aries-rfcs/tree/070b325ef4a55005d79d03eb629f0399f3a0dfc4/features/0095-basic-message), there is no way to to know if a message can contain rich text (e.g.: html, markup).

From an issuer perspective they may want to send message that contains links, and want to annotate those links in a nice hyperlink way.

Workaround:
- parse URLs based on `http://` and `http://` and display as hyperlink

**Comment by swcurran** [comment]
The “plan” amongst the early community members was that there would be a “Rich Message Protocol” to compliment the “Basic Message Protocol”.  Such a protocol was discussed at [Rebooting Web of Trust in Fall 2022](https://github.com/WebOfTrustInfo/rwot11-the-hague/blob/master/advance-readings/advanced-didcomm-messaging.md) by a group lead by @karimStekelenburg and demonstrated by (I think) @genaris that showed features like link sharing, image sharing, file sharing and so on.

I think the path to this is not to enhance Basic Message, but to define a protocol for advanced chat.

**Comment by TimoGlastra** [comment]
Agreed with what Stephen said. The protocol purposely mentions the simplicity as a feature.

@genaris has made custom protocols (and plugins for AFJ) to do read receipts, image sharing, and other rich features.
